### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by ik902a**

### CODE REVIEW

File: rest-api/src/main/java/com/epam/esm/exception_handler/IncorrectData.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package com.epam.esm.exception_handler;
```
Comment (TitenkovEA) [inline rest-api/src/main/java/com/epam/esm/exception_handler/IncorrectData.java]: dont use separators in package name, better just 'exceptionhandler'

File: rest-api/src/main/java/com/epam/esm/configuration/WebInitializer.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+	@Override
+	protected String[] getServletMappings() {
+		return new String[] { "/" };
+	}
+
+	@Override
+	public void onStartup(ServletContext servletContext) throws ServletException {
+		super.onStartup(servletContext);
+		servletContext.setInitParameter(ACTIVE_PROFILE_PARAM, PROD_PROFILE);
```
Comment (TitenkovEA) [inline rest-api/src/main/java/com/epam/esm/configuration/WebInitializer.java:36]: You have hardcoded 'prod' env, how are you going to deploy 'dev' env without changes in code?

File: dao/src/main/java/com/epam/esm/configuration/TestConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType;
+
+/**
+ * The {@code ModelConfiguration} class contains spring configuration
+ * 
+ * @author Ihar Klepcha
+ */
+@Configuration
+@ComponentScan("com.epam.esm.dao.impl")
+public class TestConfiguration {
```
Comment (TitenkovEA) [inline dao/src/main/java/com/epam/esm/configuration/TestConfiguration.java]: move this configuration to the test package

File: dao/src/main/java/com/epam/esm/service/impl/TagServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	@Override
+	@Transactional
+	public List<Tag> findAll() {
+		List<Tag> tagList = tagDao.findAll();
+		return tagList;
+	}
+
+	@Override
+	@Transactional
+	public Tag findById(long id) {//TODO fix test
```
Comment (TitenkovEA) [inline dao/src/main/java/com/epam/esm/service/impl/TagServiceImpl.java]: did you fix a test?)

File: dao/src/main/java/com/epam/esm/service/impl/GiftCertificateServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+	private GiftCertificateDao giftCertificateDao;
+	@Autowired
+	private TagDao tagDao;
+
+	@Override
+	@Transactional
+	public GiftCertificate create(GiftCertificate giftCertificate) {
+		log.info("CREATE {}", giftCertificate);
+		GiftCertificateValidator.validateGiftCertificate(giftCertificate);
+		containsName(giftCertificate.getName());// Check if DB contains Name
```
Comment (TitenkovEA) [inline dao/src/main/java/com/epam/esm/service/impl/GiftCertificateServiceImpl.java]: the comment in not clear, so can be deleted. Also I suggest you to rename the 'containsName' to 'throwErrorIfCertificateExist' or something more obvious.
P.S. this verification can be done on DB side by table constraints

**Review by TitenkovEA — CHANGES_REQUESTED** [review]
Not bad, good job! but fix the comments and also, please, remove the targets artifacts from git and add them to .gitignore

File: rest-api/src/main/java/com/epam/esm/configuration/WebInitializer.java (hunk 2 above)
Comment (ik902a) [reply rest-api/src/main/java/com/epam/esm/configuration/WebInitializer.java:36]: No way. And what to do?

**Pull request merged**